tests/neg/OrType.scala

```diff
@@ -2,7 +2,7 @@ class B(val x: Int)
 class C(val x: Double)
 
 object Test{
-  def bar(x: B | C): Int | Double = x.x
+  def bar(x: B | C): Int | Double = x.x  // error
```

Comment:
@darkdimius The previous findMember operation on union types would often fail to instantiate member types properly. I wonder whether that's an issue in the linker as well, since linker sems to use union types pervasively.